### PR TITLE
Fixed blocks being stuck in verifying state

### PR DESCRIPTION
**Pull request by arkpar**

### CODE REVIEW

File: libethereum/BlockQueue.cpp [hunk 1]
```diff
@@ -131,6 +127,7 @@ void BlockQueue::verifierBody()
 				}
 			cwarn << "BlockQueue missing our job: was there a GM?";
 			OK1:;
+			drainVerified();
```
Comment (arkpar) [inline libethereum/BlockQueue.cpp]: Here if bad block is at m_verifying[0] and m_verifying[1] contains a verified block latter will be never removed from m_verifying blocking the queue.

Comment (gavofyork) [inline libethereum/BlockQueue.cpp]: what if the bad block, `m_verifying[0]`, was on a different chain (perhaps it's a sibling) to `m_verifying[1]`?

File: libethereum/BlockQueue.cpp [hunk 2]
```diff
@@ -179,6 +167,21 @@ void BlockQueue::verifierBody()
 	}
 }
 
+void BlockQueue::drainVerified()
```
Comment (gavofyork) [inline libethereum/BlockQueue.cpp]: functions that operate on the assumption there is one or more mutexes already locked should include that in their name. e.g. `drainVerified_WITH_BOTH_LOCKS`

File: libethereum/BlockQueue.cpp [hunk 3]
```diff
@@ -179,6 +167,21 @@ void BlockQueue::verifierBody()
...
+{
+	while (m_verifying.size() && !m_verifying.front().blockData.empty())
```
Comment (gavofyork) [inline libethereum/BlockQueue.cpp]: language sanitizers generally prefer the always-fastest `!m_verifying.empty()` over `m_verifying.size()`.

File: libethereum/BlockQueue.cpp (hunk 1 above)
Comment (gavofyork) [inline libethereum/BlockQueue.cpp]: prefer this to be called `purgeBadFromVerified`.

Comment (arkpar) [inline libethereum/BlockQueue.cpp]: That would be misleading. The main purpose of this function is to clear the front of m_verifying of the blocks that have been verified, no matter good or bad. Note that is is also used on line 150 for the case of good blocks.

**Pull request merged**